apigen-deps/src/main/java/com/distelli/graphql/ResolverDataFetcher.java

```diff
@@ -37,16 +50,19 @@ public ResolverDataFetcher(DataFetcher fetcher, Resolver resolver, int listDepth
     public Object get(DataFetchingEnvironment env) {
         List<Object> unresolved = new ArrayList<>();
         Object result;
-        int depth = listDepth;
+        final int depth = listDepth;
         if ( env.getSource() instanceof List ) { // batched.
             result = getBatched(env);
-            if ( null != resolver ) addUnresolved(unresolved, result, ++depth);
+            if (nonNull(resolver) || nonNull(asyncResolver)) addUnresolved(unresolved, result, depth + 1);
         } else {
             result = getUnbatched(env);
-            if ( null != resolver ) addUnresolved(unresolved, result, depth);
+            if (nonNull(resolver) || nonNull(asyncResolver)) addUnresolved(unresolved, result, depth);
         }
-        if ( null == resolver ) return result;
-        return replaceResolved(result, resolver.resolve(unresolved).iterator(), depth);
+        if (isNull(resolver) && isNull(asyncResolver)) return result;
+        return isNull(asyncResolver) ?
+          replaceResolved(result, resolver.resolve(unresolved).iterator(), depth) :
+          asyncResolver.resolve(unresolved)
+            .thenApplyAsync(resolved -> replaceResolved(result, List.class.cast(resolved).iterator(), depth));
```

Comment:
Just curious if this List.class.cast() is necessary. Since the async resolver is now a CompletableFuture<List<T>>, it seems like that would not be necessary.